File: Chowdren/base/objects/capture.cpp
#include "objects/capture.h"

void CaptureObject::set_filename(const std::string & filename)
{
    this->filename = filename;
}

void CaptureObject::capture_frame()
{
    std::cout << "Capture frame not implemented" << std::endl;
}

void CaptureObject::capture_window()
{
    std::cout << "Capture window not implemented" << std::endl;
}

void CaptureObject::set_size(int w, int h)
{
    std::cout << "Capture set size not implemented: " << w << " " << h
        << std::endl;
}

void CaptureObject::set_origin(int x, int y)
{
    std::cout << "Capture set pos not implemented: " << x << " " << y
        << std::endl;
}

File: Chowdren/chowdren/writers/extensions/Capture.py
# Copyright (c) Mathias Kaerlev 2012-2015.
#
# This file is part of Anaconda.
#
# Anaconda is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Anaconda is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Anaconda.  If not, see <http://www.gnu.org/licenses/>.

from chowdren.writers.objects import ObjectWriter

from chowdren.common import get_animation_name, to_c, make_color

from chowdren.writers.events import (StaticConditionWriter,
    StaticActionWriter, StaticExpressionWriter, make_table)

class CaptureObject(ObjectWriter):
    class_name = 'CaptureObject'
    filename = 'capture'
    static = True
    defines = ['CHOWDREN_USE_CAPTURE']

    def write_init(self, writer):
        pass

actions = make_table(StaticActionWriter, {
    0 : 'set_filename',
    4 : 'capture_window',
    5 : 'capture_frame',
    6 : 'set_origin',
    7 : 'set_size'
})

conditions = make_table(StaticConditionWriter, {
})

expressions = make_table(StaticExpressionWriter, {
})

def get_object():
    return CaptureObject
